src/BookParser.hs

```diff
@@ -270,15 +270,10 @@ preprocessChapter = \case
 preprocessSection :: MonadState BookState n => Span -> n Span
 preprocessSection = \case
   Span t -> do
-     knownSections <- gets bookStateSectionsDefined
-     let sId = SectionId t
-     if Set.member sId knownSections then do
-       sectionsReferenced %= Set.insert sId
-       return $ SectionRef sId
-     else do
-       warn $ WInvalidSection sId
-       return $ Span t
-  s -> do
+    let sId = SectionId t
+    sectionsReferenced %= Set.insert sId
+    return $ SectionRef sId
+  s      -> do
```

Comment:
```
s      -> do
  ^^^^^
```
is this your idea of beautiful alignment? (I don't really mind, you can leave this line as is)

Reply:
Yes, it was mine idea. At first it was just

```haskell
preprocessSection = \case
   Span t -> return $ SectionRef $ SectionId t
   s      -> do
```

But I forgot to remove alignment after this refactoring. Maybe I shouldn't introduce too much new alignment...